src/libsyntax/parse/parser.rs

```diff
@@ -3149,6 +3150,14 @@ impl<'a> Parser<'a> {
         maybe_whole!(self, NtArm, |x| x);
 
         let attrs = self.parse_outer_attributes()?;
+        if self.eat(&token::BinOp(token::Or))
+            && !self.sess.unstable_features.is_nightly_build() {
```

Comment:
You should use `sess.features.borrow().match_beginning_vert` here, otherwise the feature gate won't work